### PR TITLE
LE Audio: CSIP API refactor

### LINKED ISSUE #45625: LE Audio: Update CSIP API with new naming scheme
**Is your enhancement proposal related to a problem? Please describe.**
Apply the new naming scheme from https://github.com/zephyrproject-rtos/zephyr/issues/40920 to the CSIP implementation and API

**Describe the solution you'd like**
N/A

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
Updates the CSIP API to follow the same structure as VCP and MICP. 

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/45625

### CODE REVIEW

**Comment by Thalley** [comment]
Compliance won't pass due to this bug: https://github.com/zephyrproject-rtos/zephyr/issues/51259

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by alwa-nordic — DISMISSED** [review]

File: subsys/bluetooth/shell/csip_set_coordinator.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**
  * @file
- * @brief Shell APIs for Bluetooth CSIS client
+ * @brief Shell APIs for Bluetooth CSIP set coordinator
```
Comment (kruithofa) [inline subsys/bluetooth/shell/csip_set_coordinator.c:3]: you may consider updating the copyright year

File: tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/csip.sh [hunk 2]
```diff
@@ -4,9 +4,9 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-# Basic CSIS test. A set coordinator (CSIS client) connects to multiple set
-# devices (CSIS), lock thems, unlocks them and disconnects.
-SIMULATION_ID="csis_csip"
+# Basic CSIP test. A set coordinator connects to multiple set members
```
Comment (kruithofa) [inline tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/csip.sh:7]: you may consider changing the copyright year

**Comment by kruithofa** [comment]
> Compliance won't pass due to this bug: #51259

Does it mean this PR will not be merged before that bug is fixed?

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
> > Compliance won't pass due to this bug: #51259
> 
> Does it mean this PR will not be merged before that bug is fixed?

It means that we will need to either A) override the false-positive or B) rewrite the (otherwise correct code) to pass the check or C) wait for the check to be fixed or D) remove the check from CI

A) will be easiest to do for this PR.

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by alwa-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Got tired of the CI error, so refactored the code.

**Review by alwa-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by MariuszSkamra — COMMENTED** [review]
Did you consider to shorten the `coordinator` to `coord`, and maybe `member` to `mbr`

**Comment by Thalley** [comment]
> Did you consider to shorten the `coordinator` to `coord`, and maybe `member` to `mbr`

I've been asking around for abbreviations for those, as I do think they are quite long, but haven't found any good contenders for abbreviations for those 2. 

For example `coord` is used as an abbreviation for "coordinates" in Zephyr, and `mbr` is used for "master boot record". 

Ideally I'd like to use abbreviations that does not clash with other abbreviations in Zephyr.

**Review by MariuszSkamra — DISMISSED** [review]

**Comment by Thalley** [comment]
@alwa-nordic @kruithofa Please re-review :)

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
@alwa-nordic @kruithofa @MariuszSkamra You've all previously approved this. Will you please re-review?

**Review by MariuszSkamra — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**